concordium-consensus/src/Concordium/KonsensusV1/Types.hs

```diff
@@ -590,12 +615,12 @@ data BakedBlock = BakedBlock
       bbRound :: !Round,
       -- |Block epoch number.
       bbEpoch :: !Epoch,
+      -- |The genesis 'BlockHash'.
+      bbGenesis :: !BlockHash,
```

Comment:
If I am understanding the Haskell code correctly, now you even have the genesis hash twice as part of a block. Once as a field on its own, once as a field in the QC. `blockGenesis` will grab the one in the QC. When building a `BlockHeader`, `bbBlockHeader` will grab the one from `bbGenesis`. I would remove them both.

Reply:
Addressed this, please have a look and confirm that things are as expected.